### PR TITLE
LGA-32-69 - Add model for case adaptations

### PR DESCRIPTION
**Pull request by said-moj**
## What does this pull request do?

Add model for case adaptations.

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: app/models/types/languages.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
+from enum import Enum, unique
+
+
+@unique
+class Languages(str, Enum):
```
Comment (BenMillar-MOJ) [inline app/models/types/languages.py:5]: Can we add a comment stating which ISO standard this list originates from. Just so we can easily update it if/ when the standard changes.

File: app/models/types/adaptations.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from enum import Enum
+
+
+class Adaptations(str, Enum):
```
Comment (BenMillar-MOJ) [inline app/models/types/adaptations.py:4]: Can you add a comment explaining that this list comes from the CLA CHS.

File: app/models/case_adaptations.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    BaseRequest,
+    BaseResponse,
+    BaseUpdateRequest,
+)
+from app.models.types.languages import Languages
+from app.models.types.adaptations import Adaptations
+
+
+class CaseAdaptationsBase:
+    # Should this be a list of enums
```
Comment (BenMillar-MOJ) [inline app/models/case_adaptations.py:15]: This is a list of enums, and yes 👍

File: app/models/case_adaptations.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    needed_adaptations: List[Adaptations] = Field(sa_type=JSON, nullable=True)
+    # Should this be a list of enums
```
Comment (BenMillar-MOJ) [inline app/models/case_adaptations.py:17]: This is a list of enums, and yes 👍

**Review by BenMillar-MOJ — CHANGES_REQUESTED** [review]
Looks really good, have requested some minor changes.

**Comment by BenMillar-MOJ** [comment]
The publish documentation job is failing. This is my bad, we have a PR being raised to fix this 👍

File: app/models/types/adaptations.py (hunk 2 above)
Comment (said-moj) [reply app/models/types/adaptations.py:4]: Added

File: app/models/case_adaptations.py (hunk 3 above)
Comment (said-moj) [reply app/models/case_adaptations.py]: Removed superfluous comment

File: app/models/types/languages.py (hunk 1 above)
Comment (said-moj) [reply app/models/types/languages.py:5]: Added: `List of ISO 639-1 language codes.`

File: app/db/migrations/versions/c4b9d0057513_case_adaptations.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        sa.Column("updated_at", sa.DateTime(), nullable=False),
+        sa.Column("id", sa.Uuid(), nullable=False),
+        sa.Column("needed_adaptations", sa.JSON(), nullable=True),
+        sa.Column("languages", sa.JSON(), nullable=True),
+        sa.Column("case_id", sa.Uuid(), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["case_id"],
+            ["cases.id"],
+        ),
+        sa.PrimaryKeyConstraint("id"),
```
Comment (BenMillar-MOJ) [inline app/db/migrations/versions/c4b9d0057513_case_adaptations.py:36]: The order these run in dictates the order of the columns in the table.

You may wish to manually re-order these if you wish.

**Review by BenMillar-MOJ — APPROVED** [review]
LGTM, Have left a comment 👍

**Review by TawneeOwl — APPROVED** [review]
LGTM

**Pull request merged**